DAL/EmployeeDAL.cs

```diff
@@ -0,0 +1,218 @@
+    public bool Delete(string empNo)
+    {
+        List<Employee> existingEmployees = jsonUtils.ReadJSON<Employee>(filePath);
+        Employee employeeToDelete = existingEmployees.FirstOrDefault(employee => employee.EmpNo == empNo);
+        if (employeeToDelete != null)
+        {
+            existingEmployees.Remove(employeeToDelete);
+            jsonUtils.WriteJSON(existingEmployees, filePath);
+            return true;
+        }
+        else
+        {
+            return false;
+        }
+    }
+
+    private static EmployeeDetails ConvertToEmployeeDetails(Employee employee)
+    {
+
+        EmployeeDetails employeeDetails = new()
+        {
+            EmpNo = employee.EmpNo,
+            StatusId = employee.StatusId,
+            StatusName = ((Status)employee.StatusId).ToString(),
+            FirstName = employee.FirstName,
+            LastName = employee.LastName,
+            Dob = employee.Dob,
+            Email = employee.Email,
+            MobileNumber = employee.MobileNumber,
+            LocationId = employee.LocationId,
+            LocationName = employee.LocationId.HasValue ? ((Location)employee.LocationId).ToString() : "--",
+            JoiningDate = employee.JoiningDate,
+            JobTitle = employee.JobTitle,
+            DepartmentId = employee.DepartmentId,
+            DepartmentName = employee.DepartmentId.HasValue ? ((Department)employee.DepartmentId).ToString() : "--",
+            AssignManager = employee.AssignManager,
+            AssignProject = employee.AssignProject
+        };
+        return employeeDetails;
+    }
+
+    public (List<EmployeeDetails>? results, bool status) SearchOrFilter(List<string>? keywords = null, EmployeeFilters? filters = null)
+    {
+        List<Employee> employees = jsonUtils.ReadJSON<Employee>(filePath);
+
+        List<EmployeeDetails> employeeDetailsList = employees.Select(ConvertToEmployeeDetails).ToList();
+
+        // Search logic
+        if (keywords != null && keywords.Any())
+        {
+            List<EmployeeDetails> foundEmployees = employeeDetailsList
+                .Where(employee =>
+                    keywords.Any(kw =>
+                        EmployeeContainsSearchTerm(employee, kw)
+                    )
+                ).ToList();
+
+            if (foundEmployees.Count == 0)
+            {
+                return (null, false);
+            }
+
+            return (foundEmployees, true);
+        }
+
+        // Filter logic
+        if (filters != null)
+        {
+            if (filters.Alphabet != null && filters.Alphabet.Count != 0)
+            {
+                employeeDetailsList = employeeDetailsList
+                    .Where(e =>
+                        e.FirstName != null && filters.Alphabet.Contains(char.ToLower(e.FirstName[0])))
+                    .ToList();
+            }
+            if (filters.Locations != null && filters.Locations.Count != 0)
+            {
+                employeeDetailsList = employeeDetailsList
```

Comment:
For filtering locations/depatments - inputs should be corresponding integers. Not strings.